core/src/main/java/org/infinispan/notifications/cachelistener/NotifyHelper.java

```diff
@@ -14,54 +17,73 @@
-         }
-      } else {
-         // Notify entry event after container has been updated
-         if (created) {
-            notifier.notifyCacheEntryCreated(
-                  entry.getKey(), entry.getValue(), entry.getMetadata(), false, ctx, command);
+               functionalNotifier.notifyOnWrite(() -> EntryViews.noValue(entry.getKey()));
+            }
+         } else {
+            // Notify entry event after container has been updated
+            if (created) {
+               notifier.notifyCacheEntryCreated(
+                     entry.getKey(), entry.getValue(), entry.getMetadata(), false, ctx, command);
 
-            // A write-only command only writes and so can't 100% guarantee
-            // that an entry has been created, so only send create event
-            // when the command is read-write.
-            if (!isWriteOnly)
-               functionalNotifier.notifyOnCreate(EntryViews.readOnly(entry));
+               // A write-only command only writes and so can't 100% guarantee
+               // that an entry has been created, so only send create event
+               // when the command is read-write.
+               if (!isWriteOnly)
+                  functionalNotifier.notifyOnCreate(EntryViews.readOnly(entry));
 
-            functionalNotifier.notifyOnWrite(() -> EntryViews.readOnly(entry));
-         } else {
-            notifier.notifyCacheEntryModified(entry.getKey(), entry.getValue(), entry.getMetadata(), previousValue,
-                  previousMetadata, false, ctx, command);
+               functionalNotifier.notifyOnWrite(() -> EntryViews.readOnly(entry));
+            } else {
+               notifier.notifyCacheEntryModified(entry.getKey(), entry.getValue(), entry.getMetadata(), previousValue,
+                     previousMetadata, false, ctx, command);
 
-            // A write-only command only writes and so can't 100% guarantee
-            // that an entry has been created, so only send modify when the
-            // command is read-write.
-            if (!isWriteOnly)
-               functionalNotifier.notifyOnModify(
-                  EntryViews.readOnly(entry.getKey(), previousValue, previousMetadata),
-                  EntryViews.readOnly(entry));
+               // A write-only command only writes and so can't 100% guarantee
+               // that an entry has been created, so only send modify when the
+               // command is read-write.
+               if (!isWriteOnly)
+                  functionalNotifier.notifyOnModify(
+                        EntryViews.readOnly(entry.getKey(), previousValue, previousMetadata),
+                        EntryViews.readOnly(entry));
 
-            functionalNotifier.notifyOnWrite(() -> EntryViews.readOnly(entry));
+               functionalNotifier.notifyOnWrite(() -> EntryViews.readOnly(entry));
+            }
+         }
+      }
+   }
+
+   private static Flag extractStateTransferFlag(InvocationContext ctx, FlagAffectedCommand command) {
```

Comment:
This was actually copied from [1]. I want to have it one place, but I couldn't decide where yesterday. The more I think about this morning I am thinking that NotifyHelper should be fine until we can find a better place, if any. My other thought was maybe on [2], but it seems oddly specific for it. WDYT?

[1] https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java#L550
[2] https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/context/impl/FlagBitSets.java